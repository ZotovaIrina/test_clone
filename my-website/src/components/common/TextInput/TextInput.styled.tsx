import styled from 'styled-components';
import {IFormCell} from '../FormCell/FormCell';

export const TextInputStyled = styled.input<IFormCell>`
font-family: 'RegularFont', sans-serif;
font-size: 14px;
line-height: 1.5;
border: 1px solid ${props => props.errorText ? props.theme.colors.secondary : props.theme.colors.superDarkPrimary};
padding: ${props => props.theme.spacing.gapValue / 2}${props => props.theme.spacing.gapUnit};
border-radius: 2px;
&:focus {
outline:0;
border: 2px solid ${props => props.errorText ? props.theme.colors.secondary : props.theme.colors.superDarkPrimary};
}
`;