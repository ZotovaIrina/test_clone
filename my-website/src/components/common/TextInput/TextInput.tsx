import * as React from 'react';
import FormCell, {IFormCell} from '../FormCell/FormCell';
import {useEffect, useRef, useState} from 'react';
import {TextInputStyled} from './TextInput.styled';

export interface ITextInputProps {
  formCell: IFormCell;
  inputValue: string | null;
  autoFocus?: boolean;
  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
  onChange: (option: string | null) => void;
}

const TextInput: React.FC<ITextInputProps> = props => {

  const [inputValue, setValue] = useState(props.inputValue || '');
  const textInput = useRef<HTMLInputElement>(null);

  const doAutoFocus = (autofocus: boolean | undefined) => {
    if(textInput.current && autofocus) {
      textInput.current.focus()
    }
  };

  doAutoFocus(props.autoFocus);

  useEffect(() => {
    setValue(props.inputValue || '');
  }, [props.inputValue]);

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value === '' ? null : e.target.value;
    props.onChange(value);
    setValue(e.target.value)
  };

    return (
      <FormCell {...props.formCell}>
        <TextInputStyled
          type={'text'}
          errorText={props.formCell.errorText}
          ref={textInput}
          onChange={onChange}
          onBlur={props.onBlur}
          placeholder={props.formCell.placeHolder}
          readOnly={props.formCell.disabled}
          value={inputValue}
        />
      </FormCell>
    );
};

export default TextInput;
