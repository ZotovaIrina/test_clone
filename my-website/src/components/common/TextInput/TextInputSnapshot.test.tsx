import * as React from 'react';
import { create } from 'react-test-renderer';

import TextInput from './TextInput';

it('renders correctly text input without label and error message', () => {
  const tree = create(<TextInput inputValue={'input value'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly required input with label and error message', () => {
  const tree = create(<TextInput inputValue={'input value'} label={'label'} isRequired={true} errorText={'Error text'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
