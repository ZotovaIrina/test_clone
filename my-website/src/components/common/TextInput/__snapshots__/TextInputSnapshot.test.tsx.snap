// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly required input with label and error message 1`] = `
<div
  className="form-cell form-cell--error"
>
  <span
    className="form-cell__label"
  >
    label *
  </span>
  <input
    className="faux-input"
    onChange={[Function]}
    placeholder=""
    type="text"
    value="input value"
  />
  <span
    className="form-cell__error"
  >
    Error text
  </span>
</div>
`;

exports[`renders correctly text input without label and error message 1`] = `
<div
  className="form-cell"
>
  <input
    className="faux-input"
    onChange={[Function]}
    placeholder=""
    type="text"
    value="input value"
  />
</div>
`;
